File: WFNSystem.API/Models/Provision.cs
using Amazon.DynamoDBv2.DataModel;
using System.Text.Json.Serialization;

namespace WFNSystem.API.Models;

[DynamoDBTable("WFNSystem")]
public class Provision
{
    // PK = EMP#<id_empleado>
    [DynamoDBHashKey]
    [JsonPropertyName("pk")]
    public string PK { get; set; } = string.Empty;

    // SK = PROV#<tipo>#<periodo>
    [DynamoDBRangeKey]
    [JsonPropertyName("sk")]
    public string SK { get; set; } = string.Empty;

    // Id lógico interno (por referencia)
    [DynamoDBProperty]
    [JsonPropertyName("id_Provision")]
    public string ID_Provision { get; set; } = string.Empty;

    // Relación con empleado
    [DynamoDBProperty]
    [JsonPropertyName("id_Empleado")]
    public string ID_Empleado { get; set; } = string.Empty;

    // Tipo de provisión (DECIMO_TERCERO, DECIMO_CUARTO, FONDO_RESERVA, VACACIONES)
    [DynamoDBProperty]
    [JsonPropertyName("tipoProvision")]
    public string TipoProvision { get; set; } = string.Empty;

    // Periodo YYYY-MM
    [DynamoDBProperty]
    [JsonPropertyName("periodo")]
    public string Periodo { get; set; } = string.Empty;

    // Valor mensual calculado según salario
    [DynamoDBProperty]
    [JsonPropertyName("valorMensual")]
    public decimal ValorMensual { get; set; }

    // Suma total acumulada hasta este periodo
    [DynamoDBProperty]
    [JsonPropertyName("acumulado")]
    public decimal Acumulado { get; set; }

    // Valor total pagado cuando se transfiere
    [DynamoDBProperty]
    [JsonPropertyName("total")]
    public decimal Total { get; set; }

    // TRUE si ya se pagó esta provisión en el periodo correspondiente
    [DynamoDBProperty]
    [JsonPropertyName("isTransferred")]
    public bool IsTransferred { get; set; }

    // Fecha de cálculo
    [DynamoDBProperty]
    [JsonPropertyName("fechaCalculo")]
    public string FechaCalculo { get; set; } = string.Empty;
}

File: WFN.UI/src/types/provision.ts
// Types for Provision model
export interface Provision {
  pk?: string
  sk?: string
  id_Provision: string
  id_Empleado: string
  tipoProvision: TipoProvision
  periodo: string
  valorMensual: number
  acumulado: number
  total: number
  isTransferred: boolean
  fechaCalculo: string
}

export type TipoProvision =
  | 'DECIMO_TERCERO'
  | 'DECIMO_CUARTO'
  | 'FONDO_RESERVA'
  | 'VACACIONES'
  | 'IESS_PATRONAL'

export interface ProvisionCreateDTO {
  tipoProvision: TipoProvision
  periodo: string
  valorMensual: number
  acumulado?: number
  total?: number
  isTransferred?: boolean
}

export interface ProvisionUpdateDTO extends ProvisionCreateDTO {
  id_Provision?: string
}

export interface ProcesarProvisionesRequest {
  empleadoId: string
  periodo: string
}

// Extended provision with employee data for display
export interface ProvisionConEmpleado extends Provision {
  nombreEmpleado?: string
  dniEmpleado?: string
}
